compose/config/config.py

```diff
@@ -835,6 +836,21 @@ def finalize_service_volumes(service_dict, environment):
                 finalized_volumes.append(MountSpec.parse(v, normalize, win_host))
             else:
                 finalized_volumes.append(VolumeSpec.parse(v, normalize, win_host))
+
+        duplicate_mounts = []
+        mounts = [(v.external, v.internal) for v in [_v.as_volume_spec()
+                                                     if hasattr(_v, 'as_volume_spec')
+                                                     else _v for _v in finalized_volumes]]
+        for external, internal in mounts:
+            if external and list(map(itemgetter(0), mounts)).count(
+                    external) > 1 or list(map(itemgetter(1), mounts)).count(internal) > 1:
+                duplicate_mounts.append("%s:%s" % (external, internal))
+
+        if duplicate_mounts:
+            log.error("Duplicate mount points: [%s]" % (
+                ', '.join(duplicate_mounts)))
+            sys.exit(1)
```

Comment:
You can remove the `log` and `sys.exit` calls and `raise ConfigurationError` instead,